src/common/Ellipsis/Ellipsis.js

```diff
@@ -45,24 +72,58 @@ class Ellipsis extends React.PureComponent {
 
     this.state = {
       isActive: false,
+      textContent: null,
     };
 
     this.ref = React.createRef();
   }
 
+  /**
+   * Once text component has rendered,
+   * Update text content and tooltip active state
+   * @private
+   */
+  _textDidUpdate = () => {
+    const { isActive, textContent } = this.state;
+    const newState = {};
+
+    const newTextContent = this._getTextContent();
+    if (newTextContent !== textContent) {
+      newState.textContent = newTextContent;
+    }
+
+    const shouldBeActive = this._checkShouldBeActive();
+    if (shouldBeActive !== isActive) {
+      newState.isActive = shouldBeActive;
+    }
+
+    const stateChanged = Object.keys(newState).length > 0;
+    if (stateChanged) {
+      this.setState(newState);
+    }
```

Comment:
Variable is not used elsewhere:
My suggestion:
```suggestion
    if (Object.keys(newState).length > 0) {
      this.setState(newState);
    }
```

Reply:
Don't you think it's more readable this way?
I know we don't need the variable.

Reply:
It's a simple condition in my opinion, I find it pretty readable.